docs/parameters.md

```diff
@@ -0,0 +1,324 @@
+  planName: default
+  parameters:
+    - name: txt
+      value: Plain text, passed as a string
+    - name: meta
+      # Explicitly tells that the value is a valid JSON which should be 
+      # passed as a JSON object to the broker
+      type: json
+      valueFrom:
+        secretKeyRef:
+          name: mysecret
+          key: credentials
+```
+
+#### Other primitive types (int, bool, float)
+
+```yaml
+  ...
+  parameters:
+    - name: text
+      value: 1 # string by default, will be sent as "1"
+    - name: number
+      type: int
+      value: 1 # will be sent without quotes
+    - name: flag
+      type: bool
+      value: true
+    - name: fl
+      type: float
+      value: 123.5
+```
+
+#### Arrays
+A separate field `array: true` can be used to declare a value as an array. The
+element type is defined by the `type` value as described above.
+
+```yaml
+  ...
+  parameters:
+    - name: tags
+      array: true
+      value:
+      - foo
+      - bar
+    - name: jsonArray
+      type: json
+      array: true
+      value: [{"a": 1, "b": 2}]
+```
+
+### Passing all parameters from a single source at once
+
+In addition to support for explicitly specifying every parameter value fetched from 
+a particular source, Service Catalog also supports passing multiple parameters at 
+once using a `parametersFrom` field (similar to how `envFrom` works for `Pod`s).
+Instead of specifying a reference to a single `Secret` (or `ConfigMap`) key, you 
+just set the reference to the `Secret` (or `ConfigMap`), and all keys will be 
+automatically processed as separate parameters.
+
+In other words, instead of 
+```yaml
+  ...
+  parameters:
+    - name: key1
+      valueFrom:
+        secretKeyRef:
+          name: mysecret
+          key: key1
+    - name: key2
+      valueFrom:
+        secretKeyRef:
+          name: mysecret
+          key: key2
+     ...
+```
+you can just set it once in a `parametersFrom` field:
+```yaml
+  ...
+  parametersFrom:
+    - secretRef:
+        name: mysecret
+```
+
+You can also pass several `Secret`s and/or `ConfigMap`s at once:
+```yaml
+  ...
+  parametersFrom:
+    - secretRef:
+        name: secret1
+    - secretRef:
+        name: secret2
+    - configMapRef:
+        name: configmap1
+```
+
+Parameter values could be passed as JSON objects, the same way as for [individual parameters](#json-object):
+```yaml
+  ...
+  parametersFrom:
+    - type: json
+      secretRef:
+        name: mysecret
+```
+
+### Conflict resolution
+
+Within the scope of a `parameters` or a `parametersFrom` block, if there are 
+duplicate top-level names then the last one in each section will take precedence.
+If there are duplicate top-level names across `parameters` and `parametersFrom` 
+blocks then the `parameters` always takes precedence over `parametersFrom`.
+
+This conflict resolution strategy is consistent with the one supported by Kubernetes
+for environment variables (`env` and `envFrom` fields) in the `Pod` resource.
```

Comment:
While I do like consistency I think the comparison with env vars isn't quite right. Env vars can some from multiple sources at many points in the processing chain - the OS, the image, cmd line, etc.. and so the notion of "last one wins" is appropriate, especially when you consider that no one spot in the processing has total control over the others. IOW, values are coming from places the user may not have control over and they need a way to override them.

In this case, however, we have total control and I think that makes it different from env vars. In this case there are no other params we need to merge with other than what the user is providing us. For that reason if they do provide us with a conflict then, IMO, they didn't realize their mistake and should be forced to clear up the ambiguity.

Aside from the consistency aspect, I'd like to understand the usecase for allowing someone to specify conflicting information and for us to just ignore the user's error.  Might be worth pointing out that podpresets will fail to modify a pod if there's conflict in the PP.